cirq/circuits/circuit.py

```diff
@@ -271,6 +271,23 @@ def __add__(self, other):
         result = self.copy()
         return result.__iadd__(other)
 
+    def __radd__(self, other):
+        if not isinstance(other, type(self)):
+            if not isinstance(other, (ops.Operation, Iterable)):
+                return NotImplemented
+            # Auto wrap OP_TREE inputs into a circuit.
+            other = Circuit(other)
+
+        device = (self._device if other.device is devices.UNCONSTRAINED_DEVICE
+                  else other.device)
+        device_2 = (other.device if self._device is devices.UNCONSTRAINED_DEVICE
+                    else self._device)
+        if device != device_2:
+            raise ValueError("Can't add circuits with incompatible devices.")
+
+        result = self.copy()
+        return other.__iadd__(result)
```

Comment:
This will append the circuit to the moment. Copy `other` instead so that the incremental add is performed on the copy instead of the moment or the circuit.
`
result = other.copy()
return result.__iadd__(self)
`

Reply:
Thanks a lot for your kind instructions. It's done now.